Update PII handling advice in security section (#346)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

# Privacy Considerations

Interactions with Transparency Services are expected to use appropriately strong encryption and authorization technologies.

The Transparency Service is trusted with the confidentiality of the Signed Statements presented for Registration.

> Issuers are responsible for verifying that the Transparency Service's privacy and security posture is suitable for the contents of their Signed Statements prior to Registration.

Reviewer: ```suggestion
Clients are responsible for verifying that the Transparency Service's privacy and security posture is suitable for the contents of the Signed Statements they submit prior to Registration.
```

We draw a distinction between Clients and Issuers elsewhere, and even clarify "The Client (which is not necessarily the Issuer)", and the submission performed by the Client.

I don't think we say anything about the Client always acting faithfully/honestly on behalf of the Issuer. If that's always the case, then the current wording is fine. If not, perhaps this paragraph and the next ought to be reworded slightly, and a comment added to clarify that the Client is trusted in this process?

Or maybe I am overthinking this, and this is in fact clear to everyone :)

Reviewer 2: There is a nuance between them, but I agree that Clients interact with the Transparency Service, and Issuers sign statements. Or, rather an Issuers signing material is used to sign statements, which "Clients" may operate upon.

Author: This introduces a 'turtles' situation where "Issuers are responsible for verifying that Clients that submit Signed Statements on their behalf are capable to evaluate the Transparency Service's privacy and security posture and only submit where such posture is suitable for the contents of the Signed Statements."

Reviewer: @JAG-UK quite right, I have amended the proposed wording very slightly to say "Turtles", rather than "the top-most turtle". I am also happy to see this closed without further action.

In particular, Issuers must carefully review the inclusion of private, confidential, or personally identifiable information (PII) in their Statements against the Transparency Service's privacy posture.

In some deployments a special role such as an Auditor might require and be given access to both the Transparency Service and related Adjacent Services.

> In most Transparency Service implementations the append-only log only carries cryptographic metadata (eg a hash) rather than the complete Signed Statement and so does not raise immediate privacy concerns.

Reviewer 3: Since we do not have so many instances and implementations, can we stipulate "In most..."? I don't have a strong objection, but maybe use specification language?

```suggestion
Transparency Services' append-only logs MAY carry only cryptographic metadata (e.g. a hash), rather than the complete Signed Statement, which does not raise immediate privacy concerns.
```

The Transparency Service is trusted with the confidentiality of the Signed Statements presented for Registration.
Issuers are responsible for verifying that the Transparency Service's privacy and security posture is suitable for the contents of their Signed Statements prior to Registration.
In particular, Issuers must carefully review the inclusion of private, confidential, or personally identifiable information (PII) in their Statements against the Transparency Service's privacy posture.

In some deployments a special role such as an Auditor might require and be given access to both the Transparency Service and related Adjacent Services.

In most Transparency Service implementations the append-only log only carries cryptographic metadata (eg a hash) rather than the complete Signed Statement and so does not raise immediate privacy concerns. 
By analyzing the relationship between data stored in the Transparency Service and data stored in Adjacent Services, it is possible to perform metadata analysis, which could reveal the order in which artifacts were built, signed and uploaded.

> 

Reviewer 2: ```suggestion
```

# Privacy Considerations

All contents exchanged between actors is protected using secure authenticated channels (e.g., TLS) but this does not necessarily exclude network traffic analysis.

> The Transparency Service is trusted with the confidentiality of the Signed Statements presented for Registration. Issuers are responsible for verifying that the privacy and security posture of the Transparency Service is suitable for the contents of their Signed Statements prior to Registration. In particular, Issuers must carefully review the inclusion of private, confidential, or personally identifiable information (PII) in their Statements against the privacy posture of the Transparency Service.

Reviewer 2: ```suggestion
The Transparency Service is trusted with the confidentiality of the Signed Statements presented for Registration.
Issuers are responsible for verifying that the Transparency Service's privacy and security posture is suitable for the contents of their Signed Statements prior to Registration.
In particular, Issuers must carefully review the inclusion of private, confidential, or personally identifiable information (PII) in their Statements against the Transparency Service's privacy posture.
```

# Privacy Considerations

> All contents exchanged between actors is protected using secure authenticated channels (e.g., TLS) but this does not necessarily exclude network traffic analysis.

Author: These were not my original words, I'm happy to remove altogether

> The Transparency Service is trusted with the confidentiality of the Signed Statements presented for Registration. Issuers are responsible for verifying that the privacy and security posture of the Transparency Service is suitable for the contents of their Signed Statements prior to Registration. In particular, Issuers must carefully review the inclusion of private, confidential, or personally identifiable information (PII) in their Statements against the privacy posture of the Transparency Service.

Reviewer 3: What in the architecture and specification require a TS "be trusted with" confidentiality? I can understand integrity, but not confidentiality. Secondly, is that a directly relevant concern to privacy claims, and not security?

Reviewer 4: The architecture says that the signed statement is processed by the transparency service.
You could imagine various forms of processing that might breach the expected confidentiality.
Although you might also expect this all to be handled in a public git repo, with comments and discussions on the signed statement and related policies.

Author: I'd be happy to remove the first sentence. They're not my original words and I think the following text covers the whole topic better than that headline.

> This document supports opaque references (such as hashes) for cases where information needs to be included on the log whilst remaining secret.

Reviewer 4: ```suggestion
Using a hash as the statement about an artifact, supports cases where information needs to be included on the append-only log whilst remaining opaque to the transparency service.
```

Author: Not a reference to Hash Envelope specifically, just pointing out that people can commit to data with disclosing it.

I like your suggestion and will commit it.

> While not strictly required for interoperability of Receipts and Signed Statement Registration, most Transparency Services are expected to keep copies of Registered Signed Statements in an auxiliary service nd make them available to authorized users for ease of indexing, auditing or similar activities.

Author: Good idea should add that out-of-scope comment

> While not strictly required for interoperability of Receipts and Signed Statement Registration, most Transparency Services are expected to keep copies of Registered Signed Statements in an auxiliary service nd make them available to authorized users for ease of indexing, auditing or similar activities.

Reviewer 2: Agreed to make out of scope. I also noticed the wording says "...expected to keep copies..." which is also not necessarily true. "...may keep copies..."

> Further, in some implementations of Transparency Services such Signed Statements may be made available publicly for dissemination, wider auditing, or to support query confidentiality. In such cases Issuers are responsible for ensuring that Signed Statements submitted to public access services are acceptable for public disclosure.

Reviewer 4: Are we talking about submitted a signed statement to an artifact repository here?

Or submitting a signed statement to public transparency service?

Author: Either / both? Not sure it matters. I think these were Hannes' words originally, I'm just trying to tidy up the Considerations section so moved things around. 

TBH the public reference is just really over-explaining the point above it, we could consider removing, but I'm sure others would insist on putting it back again

> In some deployments a special role such as an Auditor might require and be given access to Signed Statements.

Reviewer 4: ```suggestion
In some deployments a special role such as an Auditor might require and be given access to both the Transparency Service and related Adjacent Services.
```

> Once a Signed Statement has been Registered with a Transparency Service its log entry cannot be removed. In most Transparency Service implementations the append-only log only carries cryptographic metadata (eg a hash) rather than the complete Signed Statement and so does not raise immediate privacy concerns. Nonetheless even this hash can be used to hold an Issuer accountable for statements if an adversary has a copy of the pre-image and the log data, and the number of log entries registered can also give clues about Issuer activity. Such side-channel signals are out of scope for this document to deal with, but should be considered in sensitive use cases.

Reviewer 4: ```suggestion
In most Transparency Service implementations the append-only log only carries cryptographic metadata (eg a hash) rather than the complete Signed Statement and so does not raise immediate privacy concerns. 
By analyzing the relationship between data stored in the Transparency Service and data stored in Adjacent Services, it is possible to perform metadata analysis, which could reveal the order in which artifacts were built, signed and uploaded.
```

Author: I like this change, but the original text that I ws working with specifically wanted to very clearly spell out that you can't remove data from an append-only log (IKR 🤷🏻‍♂️) so I stuck with that angle. 

Let's discuss in the group how naive a reader we expect here.

> Care should be taken by Issuers to ensure that linking Signed Statements into a collection does not leak information about the contents of the other Signed Statements referred to.

Reviewer 4: ```suggestion
```

Reviewer 4: I think this is covered by the comment on linking TS and AS (adjacent service) data.

> The confidentiality of queries is implementation-specific, and generally not guaranteed.

Reviewer 4: ```suggestion
```

The architecture does not specify a query interface... SCRAPI does though.

> For example, while offline Envelope validation of Signed Statements is private, a Transparency Service may monitor which of its Transparent Statements are being verified from lookups to ensure their freshness.

Reviewer 4: ```suggestion
```

I think this is covered by the comment regarding metadata analysis.

This sentence might be more appropriate in SCRAPI.

Author: Should discuss, as the append-on-log itself is also valid here

# Privacy Considerations

> Interactions with Transparency Services MUST use appropriately strong encryption and authorization technologies.

Reviewer 2: ```suggestion
Interactions with Transparency Services are expected to use appropriately strong encryption and authorization technologies.
```

### Confidentiality and Privacy

All contents exchanged between actors is protected using secure authenticated channels (e.g., TLS) but this does not necessarily exclude network traffic analysis.

The Transparency Service is trusted with the confidentiality of the Signed Statements presented for Registration.

Some Transparency Services may make every Signed Statement registered with them public, for example to facilitate their dissemination and auditing.

> Issuers are responsible for verifying that the privacy and security posture of the Transparency Service is suitable for the contents of their Signed Statements prior to Registration. In particular, Issuers must carefully review the inclusion of private, confidential, or personally identifiable information (PII) in their Statements against the privacy posture of the Transparency Service.

Reviewer 2: ```suggestion
Issuers are responsible for verifying that the privacy and security posture of the Transparency Service is suitable for the contents of their Signed Statements prior to Registration.
In particular, Issuers must carefully review the inclusion of private, confidential, or personally identifiable information (PII) in their Statements against the privacy posture of the Transparency Service.
```
